### PR TITLE
Ingress-nginx 0.17.1

### LINKED ISSUE #513: Upgrade ingress-nginx to 0.17.x

### PR DESCRIPTION
**Pull request by jakolehm**
Fixes #513

### CODE REVIEW

**Comment by jakolehm** [comment]
```
kubectl logs nginx-ingress-controller-cmh2s -n ingress-nginx
-------------------------------------------------------------------------------
NGINX Ingress controller
  Release:    0.17.1
  Build:      git-12f7966
  Repository: https://github.com/kubernetes/ingress-nginx.git
-------------------------------------------------------------------------------

nginx version: nginx/1.13.12
W0809 12:09:37.656588       6 client_config.go:552] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0809 12:09:37.656874       6 main.go:191] Creating API client for https://172.32.0.1:443
I0809 12:09:37.673023       6 main.go:235] Running in Kubernetes cluster version v1.11 (v1.11.1) - git (clean) commit b1b29978270dc22fecc592ac55d903350454310a - platform linux/amd64
I0809 12:09:37.679283       6 main.go:100] Validated ingress-nginx/default-http-backend as the default backend.
I0809 12:09:38.201106       6 nginx.go:255] Starting NGINX Ingress controller
I0809 12:09:38.215368       6 event.go:221] Event(v1.ObjectReference{Kind:"ConfigMap", Namespace:"ingress-nginx", Name:"tcp-services", UID:"fcf523a6-9bcc-11e8-bcb1-a6af7d1dc91b", APIVersion:"v1", ResourceVersion:"562", FieldPath:""}): type: 'Normal' reason: 'CREATE' ConfigMap ingress-nginx/tcp-services
I0809 12:09:38.215492       6 event.go:221] Event(v1.ObjectReference{Kind:"ConfigMap", Namespace:"ingress-nginx", Name:"nginx-configuration", UID:"fcbb629a-9bcc-11e8-bcb1-a6af7d1dc91b", APIVersion:"v1", ResourceVersion:"561", FieldPath:""}): type: 'Normal' reason: 'CREATE' ConfigMap ingress-nginx/nginx-configuration
I0809 12:09:38.217869       6 event.go:221] Event(v1.ObjectReference{Kind:"ConfigMap", Namespace:"ingress-nginx", Name:"udp-services", UID:"fd2ca447-9bcc-11e8-bcb1-a6af7d1dc91b", APIVersion:"v1", ResourceVersion:"563", FieldPath:""}): type: 'Normal' reason: 'CREATE' ConfigMap ingress-nginx/udp-services
I0809 12:09:39.401790       6 nginx.go:276] Starting NGINX process
I0809 12:09:39.401977       6 leaderelection.go:185] attempting to acquire leader lease  ingress-nginx/ingress-controller-leader-nginx...
W0809 12:09:39.402364       6 controller.go:359] Service "ingress-nginx/default-http-backend" does not have any active Endpoint
I0809 12:09:39.402406       6 controller.go:169] Configuration changes detected, backend reload required.
I0809 12:09:39.412465       6 leaderelection.go:194] successfully acquired lease ingress-nginx/ingress-controller-leader-nginx
I0809 12:09:39.412530       6 status.go:197] new leader elected: nginx-ingress-controller-cmh2s
I0809 12:09:39.517353       6 controller.go:185] Backend successfully reloaded.
W0809 12:09:39.521726       6 nginx_status.go:207] unexpected error obtaining nginx status info: unexpected error scraping nginx status page: unexpected error scraping nginx : Get http://0.0.0.0:18080/nginx_status: dial tcp 0.0.0.0:18080: connect: connection refused
W0809 12:09:39.522490       6 nginx_status.go:207] unexpected error obtaining nginx status info: unexpected error scraping nginx status page: unexpected error scraping nginx : Get http://0.0.0.0:18080/nginx_status: dial tcp 0.0.0.0:18080: connect: connection refused
W0809 12:09:42.735832       6 controller.go:359] Service "ingress-nginx/default-http-backend" does not have any active Endpoint
W0809 12:09:46.069166       6 controller.go:359] Service "ingress-nginx/default-http-backend" does not have any active Endpoint
W0809 12:09:49.402483       6 controller.go:359] Service "ingress-nginx/default-http-backend" does not have any active Endpoint
W0809 12:09:53.721589       6 controller.go:359] Service "ingress-nginx/default-http-backend" does not have any active Endpoint
I0809 12:10:01.629157       6 controller.go:169] Configuration changes detected, backend reload required.
I0809 12:10:01.818775       6 controller.go:185] Backend successfully reloaded.
I0809 12:10:04.962730       6 controller.go:169] Configuration changes detected, backend reload required.
I0809 12:10:05.120100       6 controller.go:185] Backend successfully reloaded.
```

File: addons/ingress-nginx/resources/daemonset.yml.erb [hunk 1]
```diff
@@ -43,6 +33,14 @@ spec:
             - --tcp-services-configmap=$(POD_NAMESPACE)/tcp-services
             - --udp-services-configmap=$(POD_NAMESPACE)/udp-services
             - --annotations-prefix=nginx.ingress.kubernetes.io
+          securityContext:
```
Comment (jakolehm) [inline addons/ingress-nginx/resources/daemonset.yml.erb]: This change is from `0.16.0`, see: https://github.com/kubernetes/ingress-nginx/releases/tag/nginx-0.16.0

File: addons/ingress-nginx/resources/daemonset.yml.erb [hunk 2]
```diff
@@ -23,16 +23,6 @@ spec:
       <%- end -%>
       serviceAccountName: nginx-ingress-serviceaccount
       hostNetwork: true
-      initContainers:
-      - command:
-        - sh
-        - -c
-        - sysctl -w net.core.somaxconn=32768; sysctl -w net.ipv4.ip_local_port_range="1024 65535"
```
Comment (SpComb) [inline addons/ingress-nginx/resources/daemonset.yml.erb]: Is this tuning no longer relevant? It's still there in the examples: https://github.com/kubernetes/ingress-nginx/tree/master/docs/examples/customization/sysctl

https://github.com/kubernetes/ingress-nginx/pull/2077

Comment (jakolehm) [reply addons/ingress-nginx/resources/daemonset.yml.erb]: As a default it might be bad, reason why upstream removed it from the deployment manifest.

Comment (jakolehm) [reply addons/ingress-nginx/resources/daemonset.yml.erb]: Reading the issue I think we are fine... going to re-enable this.

Comment (SpComb) [reply addons/ingress-nginx/resources/daemonset.yml.erb]: Hmm... would new enough versions of kube support pod sysctls natively? https://kubernetes.io/docs/tasks/administer-cluster/sysctl-cluster/#setting-sysctls-for-a-pod

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: sysctl-example
spec:
  securityContext:
    sysctls:
    - name: kernel.shm_rmid_forced
      value: "0"
    - name: net.ipv4.route.min_pmtu
      value: "552"
    - name: kernel.msgmax
      value: "65536"
  ...
```

Comment (jakolehm) [reply addons/ingress-nginx/resources/daemonset.yml.erb]: Seems so, going to open separate issue about that.

Comment (SpComb) [reply addons/ingress-nginx/resources/daemonset.yml.erb]: Seems like `net.ipv4.ip_local_port_range` is considered a safe sysctl and allowed by default, but tuning `net.core.somaxconn` would require `kubelet --allowed-unsafe-sysctls net.core.somaxconn`.

**Comment by SpComb** [comment]
The ingress-nginx daemonset has `updateStrategy.type OnDelete`, so a `pharos-cluster up` doesn't actually upgrade the pods... needs `kubectl -n ingress-nginx delete pods -l app=ingress-nginx`. Not sure if bug or feature.

**Pull request merged**